compiler/codegen/library.cpp

```diff
@@ -89,6 +94,13 @@ static std::string getCompilelineOption(std::string option) {
   return res;
 }
 
+static void setupMakeEnvVars(std::string var, fileinfo makefile) {
+  std::string cmd = "echo $";
+  cmd += var;
+  std::string cmdRes = runCommand(cmd);
```

Comment:
I'm confused about why you added this code?  What case is it supporting?

Reply:
The values for CHPL_HOME, CHPL_RUNTIME_INCL, etc, weren't necessarily going to be in the make environment unless I put them into the generated makefile.  I had already had that line back when we only added CHPL_HOME to the top of the file, but realized it would get redundant if I just copy+pasted it for the three new variables I was using.  So I made a helper function and passed the environment variable name in instead

Reply:
I had messed up this step due to a typo for CHPL_RUNTIME_INCL, and it didn't get found in the configure environment (even though the replacement in the compileline output was valid)

Reply:
I should probably comment it though, d'oh!

Reply:
My question is more about why it runs `echo $SOMEVARIABLE`

Reply:
To get the value of SOMEVARIABLE.  runCommand executes a command via popen, which I thought was just a shell command.  `$CHPL_HOME` on its own in my terminal gives an error message about its value being a directory.  Is there a better strategy to get the value that I have missed?

Reply:
If you just want to know the value of an environment variable you can use getenv. The part that is unclear to me is - what $SOMEVARIABLE variables are present here that you're trying to get the value of?

In the PR it looks like you call this function like this:
```
  setupMakeEnvVars("CHPL_RUNTIME_LIB", makefile);
  setupMakeEnvVars("CHPL_RUNTIME_INCL", makefile);
  setupMakeEnvVars("CHPL_THIRD_PARTY", makefile);
  setupMakeEnvVars("CHPL_HOME", makefile);
```

`CHPL_RUNTIME_LIB`, `CHPL_RUNTIME_INCL`, `CHPL_THIRD_PARTY` and `CHPL_HOME` are all variables the compiler knows and uses and has C variables for (in driver.h). So I'm saying, why don't you just pass that?

e.g.
```
  setupMakeEnvVars("CHPL_RUNTIME_LIB", CHPL_RUNTIME_LIB, makefile);
```

(and not runCommand at all).

Reply:
> why don't you just pass that?

Because I didn't know they existed :) I'll update the implementation to do that, thanks for pointing it out!